src/Common/Auth/OpenIDConnect/Repositories/ScopeRepository.php

```diff
@@ -132,4 +73,323 @@ public function finalizeScopes(
 
         return $scopes;
     }
+
+    // nonce claim and nonce scope is handled by server logic.
+    // current locale is enUS need to set from openemr locale.
+    public function getSupportedClaims(): array
+    {
+        return array(
+            "profile",
+            "email",
+            "email_verified",
+            "phone",
+            "phone_verified",
+            "family_name",
+            "given_name",
+            "fhirUser",
+            "locale",
+            "api:oemr",
+            "api:fhir",
+            "api:port",
+            "api:pofh",
+            "aud", //client_id
+            "iat", // token create time
+            "iss", // token issuer(https://domain)
+            "exp", // token expiry time.
+            "sub" // the subject of token. usually patient UUID.
+        );
+    }
+
+    public function oidcScopes(): array
+    {
+        return [
+            "openid",
+            "profile",
+            "name",
+            "address",
+            "given_name",
+            "family_name",
+            "nickname",
+            "phone",
+            "phone_verified",
+            "email",
+            "email_verified",
+            "offline_access",
+            "api:oemr",
+            "api:fhir",
+            "api:port",
+            "api:pofh"
+        ];
+    }
+
+    public function fhirRequiredSmartScopes(): array
+    {
+        return [
+            "openid",
+            "fhirUser",
+            "online_access",
+            "offline_access",
+            "launch",
+            "launch/patient",
+            "api:oemr",
+            "api:fhir",
+            "api:port",
+            "api:pofh"
+        ];
+    }
+
+    /**
+     * @Method fhirScopes
+     *
+     * Returns all fhir permitted scopes and permissions.
+     * These will be qualified against existing and future routes
+     * with role(patient, user or system facing applications).
+     *
+     * @return array
+     */
+    public function fhirScopes(): array
```

Comment:
So eventually your thought is we will put these into the database and let the site administrator turn this on or off right?

Just trying to make sure I understand that these are the **permitted** scopes that we are allowing FHIR authors to make calls against versus what the system technologically supports which are defined in the REST api.  I think I'm following everything here but I just want to double check.

Reply:
Yep, they are a list of all current or future supported scopes. Where we only publish scopes the api supports at the time validated against this list.
Hopefully when we add a new Fhir resource to our routes then scopes will already be in place

Reply:
> Only thing I'm not entirely clear on is if our scopes listed in the oauth2/provider/.well-known/discovery.php need to match what we list in our SMART well-known.

hmm, I need to think about that for a minute. I'm leaning towards you are prob right and should, especially if we offer the api:fhir scope. What about standard apis? Should we limit them by scopes as we do for Fhir?

Reply:
It'd be great to keep them all consistent, but I don't fully understand all the uses cases for the standard api like you and Brady do, so I would think that'd be y'alls call.

Reply:
It may be what @bradymiller was speaking about with a custom scope for native however, since we are putting in place the permissions then perhaps we should scope standard as well. Makes no sense to limit Fhir access and not standard too.

Reply:
I'm thinking we should preface our scopes with `fhir:user/Patient.read` and `standard:user/patient.read`

Reply:
I'm not following.  ONC testing won't send any **fhir:** prefix. Would you stripe out the prefix when comparing against a SMART app registration or authorization request?

Reply:
I could but maybe instead just preface for our standard api thus leaving fhir alone. OpenEMR I think is in a spot by offering two api sets. It is just very handy for vendor apps to be able to call both or many vendors will write apps using only our standard api.
Bringing into question; do we register client apps as only Fhir or standard, public/user/system facing on a per client basis or allow one client to support several apps. A phone call please so I can explain better:)

Reply:
we can schedule a call today for 11 :)

Reply:
works for me. can you do @adunsulag